csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -38,7 +38,7 @@ namespace Apache.Arrow.Adbc.Drivers.Apache.Spark
 {
     public class SparkConnection : HiveServer2Connection
     {
-        const string UserAgent = "MicrosoftSparkODBCDriver/2.7.6.1014";
+        private readonly string UserAgent = $"{InfoDriverName.Replace(" ", "")}/{ProductVersionDefault}";
```

Comment:
Note that some servers actually change their behavior based on the reported client version -- Snowflake definitely does this -- so while this change is right in principle it may have consequences in practice.